android/src/main/java/com/rudderstack/android/ConfigurationAndroid.kt

```diff
@@ -210,7 +261,7 @@ interface ConfigurationAndroid : Configuration {
         sdkVerifyRetryStrategy: RetryStrategy,
         dataPlaneUrl: String,
         controlPlaneUrl: String?,
-        logger: Logger,
+        rudderLogger: RudderLogger,
```

Comment:
I believe, rudder logger shouldn't be a part of copy method, otherwise it will be exposed to change by the users.

Reply:
I just attempted to remove it but there I got an error: 
<img width="214" alt="image" src="https://github.com/rudderlabs/rudder-sdk-android/assets/64667840/5c2608ca-bcad-49ba-a783-466adf29b086">

Reply:
We discussed this offline and came to the conclusion that since removing the logger field from the `copy` method is not possible, we are going to restrict the user from changing this value during runtime.
CC: @itsdebs